rasa/nlu/tokenizers/character_tokenizer.py

```diff
@@ -0,0 +1,16 @@
+from typing import List, Text
+
+from rasa.nlu.tokenizers.tokenizer import Token, Tokenizer
+from rasa.shared.nlu.training_data.message import Message
+
+
+class CharacterTokenizer(Tokenizer):
+    # the following language should can be tokenized using the CharacterTokenizer
+    supported_language_list = ["zh", "ja", "th"]
+
+    def tokenize(self, message: Message, attribute: Text) -> List[Token]:
+        text = message.get(attribute)
+
+        words = [i for i in text]
```

Comment:
Can this well support the mixture of Chinese and numbers? For example, "他的手机号是12345677890", if all the numbers are turned into a single character, will it take up too much length in the pre-trained model with limited length?

Reply:
Yes, it supports such a Chinese sentence, As far as I know, the BERT model for Chinese is character-tokenized (https://github.com/google-research/bert/blob/master/multilingual.md#tokenization), all the numbers turned into individual characters is expected behavior. In the chatbot domain, people generally use short text (in my experience, the average length will be 14, and 99% of text shorter than 32), so I don't think the length of Chinese tokens will be an issue.

Reply:
Is this also true for Japanese and Thai?

Reply:
> Yes, it supports such a Chinese sentence, As far as I know, the BERT model for Chinese is character-tokenized (https://github.com/google-research/bert/blob/master/multilingual.md#tokenization), all the numbers turned into individual characters is expected behavior. In the chatbot domain, people generally use short text (in my experience, the average length will be 14, and 99% of text shorter than 32), so I don't think the length of Chinese tokens will be an issue.

I agree with you that we don’t need to worry about the length issue in conversations , but it seems that BERT character-tokenized is only for Chinese characters. For mixed strings, it only adds spaces on both sides of Chinese characters but not for numbers and English letters, and use whitespace_tokenize after that.
>> Because Chinese (and Japanese Kanji and Korean Hanja) does not have whitespace characters, we add spaces around every character in the CJK Unicode range before applying WordPiece. This means that Chinese is effectively character-tokenized. Note that the CJK Unicode block only includes Chinese-origin characters and does not include Hangul Korean or Katakana/Hiragana Japanese, which are tokenized with whitespace+WordPiece like all other languages.

https://github.com/google-research/bert/blob/master/tokenization.py#L251-L284

Reply:
We are actually working on another issue that would remove (or first deprecate) the language specific tokenizers. E.g. if you want to use the `LanuageModelFeaturizer` you would not be forced to use the `LanguageModelTokenizer` anymore, but instead you could use any avaiable tokenizer. So you could combine for example the `JiebaTokenizer` and the `LanguageModelFeaturizer`. Would that solve this problem?